flocker/testtools/_flaky.py

```diff
@@ -22,10 +37,179 @@ def flaky(jira_key):
 
     # TODO (see FLOC-3281):
     # - allow specifying which exceptions are expected
-    # - retry on expected exceptions
-    #   - parametrize on retry options
     # - provide interesting & parseable logs for flaky tests
 
+    if isinstance(jira_keys, basestring):
```

Comment:
:question: This may be a losing battle ... but ... I don't think `basestring` is what you mean here.  Based on all the byte->unicode changes in this branch, I think you mean `unicode` here :smile: At worst, I think you mean `(bytes, unicode)` though.  `basestring` used to mean "The C struct layout of this object is compatible with PyString_Type / PyUnicode_Type" and then CPython changed the C struct layout of PyUnicode_Type so it was impossible to be compatible with both so now `basestring` doesn't mean anything afaict.